pkg/vpcmodel/abstractVPC.go

```diff
@@ -8,6 +8,8 @@ import (
 type VPCResourceIntf interface {
 	UID() string
 	Name() string
+	Names() (string, []string)
+	Type() string
```

Comment:
Where does someone get the list of available type strings to compare against? Seems like they are scattered in several files.

Reply:
Added a comment as to why this is needed. Not sure I understand the 2nd sentence

Reply:
Is `Type()` meant just to be printed as-is to the drawio file, or is it going to be used programmatically to check the type of a give VPC resource? If the latter is true, how do I know what is the correct string to check against? Do I have to take a look at the code and copy the same type string to my code?

Reply:
I guess the latter is true, but this is a question to @haim-kermany: if the current interface which outputs string (I did it quick and dirty, true) is not good enough, please suggest the required interface. Do we need to introduce enums?

Reply:
Type() is needed to distinguish between kind of grouping (i.e. endpoint).
each grouping is implemented differently in the drawio.
I do not know yet what kind of  grouping will we have.
 but the current implementation of returning strings is problematic, since some of the stings are defined in the ibmvpc pkg
.

Reply:
I believe the next comment by Haim and my fix also resolves this one

Reply:
@haim-kermany your next comment is about this. I fixed by using the existing `Kind()` as you ask for - however, these uses strings defined in `ibmvpc` which you can not import.  
Perhaps move these constants to some shared library, also the new constants defined in `grouping.go`?  @zivnevo

Reply:
@haim-kermany , where exactly do you have to use these Kinds? Currently they are only used for preparing the drawio  TreeNode model in `pkg\vpcmodel\drawioOutput.go` which is part of `vpcmodel`. The new Kinds in `grouping.go` are also in `vpcmodel` so I see no circular dependencies. Would you like the new grouped types to implement the `VPCResourceIntf` interface, and especially `DetailsMap()` where you usually take the Kind from?

Reply:
I do not like DetailsMap(). it is not clear what are the keys there. I used because I do not want anything else.
Idealy,  if VPCResourceIntf could give me a better interface than DetailsMap(), it would be better.
In grouping, I have another problem. i need to know the type of EndpointElem( which is not only VPCResourceIntf )

Reply:
@haim-kermany , maybe open a new issue about how you suggest the `abstractVPC -> drawio` interface to look like. It's not to late to refactor. Maybe we can do it together with issue #141 .
For this work, I suggest to still use the `DetailsMap()`, and maybe add new keys to it to reflect the type of `EndpointElem`.

Reply:
I just relized we have muke kind of grouping that I thought.
I will be much wizer after I know what kind of grouping do we have, and if/how we will present them
will all should discuss with Adi and keren